tirocks/src/db/flush.rs

```diff
@@ -0,0 +1,252 @@
+    /// flush enabled.
+    #[inline]
+    pub fn flush_multi(&self, option: &FlushOptions, cfs: &[&RawCfHandle]) -> Result<()> {
+        unsafe {
+            let cfs: Vec<_> = cfs.iter().map(|c| c.get_ptr()).collect();
+            ffi_call!(crocksdb_flush_cfs(
+                self.get_ptr(),
+                option.as_ptr(),
+                cfs.as_ptr(),
+                cfs.len() as i32,
+            ))
+        }
+    }
+
+    /// Flush the WAL memory buffer to the file. If sync is true, it calls [`sync_wal`]
+    /// afterwards.
+    #[inline]
+    pub fn flush_wal(&self, sync: bool) -> Result<()> {
+        unsafe { ffi_call!(crocksdb_flush_wal(self.get_ptr(), sync)) }
+    }
+
+    /// Sync the wal. Note that [`write`] followed by [`sync_wal`] is not exactly the
+    /// same as [`write`] with sync=true: in the latter case the changes won't be
+    /// visible until the sync is done.
+    /// Currently only works if allow_mmap_writes = false in Options.
+    #[inline]
+    pub fn sync_wal(&self) -> Result<()> {
+        unsafe { ffi_call!(crocksdb_sync_wal(self.get_ptr())) }
+    }
+
+    /// Load a list of external SST files (1) into the DB
+    /// Two primary modes are supported:
+    /// - Duplicate keys in the new files will overwrite exiting keys (default)
+    /// - Duplicate keys will be skipped (set ingest_behind=true)
+    /// In the first mode we will try to find the lowest possible level that
+    /// the file can fit in, and ingest the file into this level (2). A file that
+    /// have a key range that overlap with the memtable key range will require us
+    /// to Flush the memtable first before ingesting the file.
+    /// In the second mode we will always ingest in the bottom most level (see
+    /// docs to IngestExternalFileOptions::ingest_behind).
+    ///
+    /// (1) External SST files can be created using SstFileWriter
+    /// (2) We will try to ingest the files to the lowest possible level
+    ///     even if the file compression doesn't match the level compression
+    /// (3) If IngestExternalFileOptions->ingest_behind is set to true,
+    ///     we always ingest at the bottommost level, which should be reserved
+    ///     for this purpose (see DBOPtions::allow_ingest_behind flag).
+    #[inline]
+    pub fn ingest_external_file(
+        &self,
+        opt: &IngestExternalFileOptions,
+        cf: &RawCfHandle,
+        files: &[impl AsRef<Path>],
+    ) -> Result<()> {
+        unsafe {
+            let files: Vec<_> = files.iter().map(|p| p.path_to_slice()).collect();
+            ffi_call!(crocksdb_ingest_external_file_cf(
+                self.get_ptr(),
+                cf.get_ptr(),
+                files.as_ptr(),
+                files.len(),
+                opt.as_ptr(),
+            ))
+        }
+    }
+
+    /// Same as [`ingest_external_file`] but avoid blocking writes with best effort.
+    #[inline]
+    pub fn ingest_external_file_optimized(
```

Comment:
How about using it as the impl of `ingest_external_file`

Reply:
but it's just an optimization without breaking any behavior

Reply:
It will flush if overlap.